oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java

```diff
@@ -129,4 +131,16 @@ protected String getTableQuery(DBConnectorPath path) {
   protected String getTableQuery(DBConnectorPath path, int limit) {
     return String.format("SELECT * FROM %s.%s WHERE ROWNUM <= %d", path.getSchema(), path.getTable(), limit);
   }
+
+  @Override
+  protected Schema getSchema(int sqlType, String typeName, int scale, int precision, String columnName,
+                             boolean handleAsDecimal) throws SQLException {
+    // For a Number type without specified precision and scale, precision will be 0 and scale will be -127
+    if (precision == 0) {
+      // reference : https://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm#CNCPT1832
+      precision = 38;
```

Comment:
Do you know how the source handles this scenario, will the pipeline run fail or transform the record wrong?

Reply:
source will fail. I have the fix for source in this PR too.

Reply:
OracleSourceSchemaReader and OracleSourceDBRecord are used by both connector and source